__tests__/Dict/selectKeys.test.ts

```diff
@@ -0,0 +1,65 @@
+import { expectType } from 'ts-expect'
+
+import { D, pipe } from '../..'
+
+const obj = { x: 123, y: 'abc', z: false }
+
+const rec: Record<string, number> = { a: 1, b: 2, c: 3 }
+
+describe('selectKeys', () => {
+  it('provides correct types', () => {
+    expectType<Pick<typeof obj, 'x' | 'y'>>(D.selectKeys(obj, ['x', 'y']))
+    expectType<typeof rec>(D.selectKeys(rec, ['a', 'b']))
+    expectType<typeof rec>(D.selectKeys(rec, ['y', 'z']))
+  })
+
+  it('selects the right keys', () => {
+    expect(D.selectKeys(obj, ['x', 'y'])).toEqual({ x: 123, y: 'abc' })
+    expect(D.selectKeys(rec, ['a', 'b'])).toEqual({ a: 1, b: 2 })
+    expect(D.selectKeys(rec, ['y', 'z'])).toEqual({})
+  })
+
+  it('*', () => {
+    expect(
+      D.selectKeys({ name: 'Joe', location: 'Warsaw', age: 42 }, [
+        'name',
+        'age',
+      ]),
+    ).toEqual({ name: 'Joe', age: 42 })
+    expect(
+      D.selectKeys(
+        { name: 'Joe', location: 'Warsaw' } as Record<string, string>,
+        ['name', 'age'],
+      ),
+    ).toEqual({ name: 'Joe' })
+  })
+})
+
+describe('selectKeys (pipe)', () => {
+  it('provides correct types', () => {
+    expectType<Pick<typeof obj, 'x' | 'y'>>(pipe(obj, D.selectKeys(['x', 'y'])))
+    expectType<typeof rec>(pipe(rec, D.selectKeys(['a', 'b'])))
+    expectType<typeof rec>(pipe(rec, D.selectKeys(['y', 'z'])))
+  })
+
+  it('selects the right keys', () => {
+    expect(pipe(obj, D.selectKeys(['x', 'y']))).toEqual({ x: 123, y: 'abc' })
+    expect(pipe(rec, D.selectKeys(['a', 'b']))).toEqual({ a: 1, b: 2 })
+    expect(pipe(rec, D.selectKeys(['y', 'z']))).toEqual({})
+  })
+
+  it('*', () => {
+    expect(
+      pipe(
+        { name: 'Joe', location: 'Warsaw', age: 42 },
+        D.selectKeys(['name', 'age']),
+      ),
+    ).toEqual({ name: 'Joe', age: 42 })
+    expect(
+      pipe(
+        { name: 'Joe', location: 'Warsaw' } as Record<string, string>,
```

Comment:
is `Record<string, string>` necessary here? if so, I feel we need to work on types and improve them somehow, I will look into it :)

Reply:
Hi @mobily! What I'm trying to get at here is that if the type of `{ name: 'Joe', location: 'Warsaw'}` is `{ name: string, location: string}`, selecting `age` would not type check. This situation can arise if you are dealing with records - if the key is not present, you should not get anything selected. Perhaps the intent of the test is not super clear! Please let me know how you'd like to change it!

Reply:
ahh, now I got it, thanks for the explanation! ❤️ you may want to check out how I handle this case:
https://github.com/mobily/ts-belt/blob/master/__tests__/Dict/update.test.ts#L115